src/GoogleAdWordsEventForwarder.js

```diff
@@ -124,23 +121,26 @@
             return 'Can\'t send to forwarder ' + name + ', not initialized. Event added to queue.';
         }
 
-        function setEnhancedConversionData(enhancedConversionData) {
-            if (enhancedConversionData.email) {
-                window.enhanced_conversion_data.email = enhancedConversionData.email;
-            }
-            if (enhancedConversionData.phone_number) {
-                window.enhanced_conversion_data.phone_number = enhancedConversionData.phone_number;
-            }
-            if (enhancedConversionData.first_name) {
-                window.enhanced_conversion_data.first_name = enhancedConversionData.first_name;
-            }
-            if (enhancedConversionData.last_name) {
-                window.enhanced_conversion_data.last_name = enhancedConversionData.last_name;
-            }
-            if (enhancedConversionData.home_address) {
-                window.enhanced_conversion_data.home_address =
-                    enhancedConversionData.home_address;
+        function hasEnhancedConversionData(customFlags) {
+            return customFlags && Object.keys(customFlags).length && customFlags[ENHANCED_CONVERSION_DATA];
+        }
+
+        function parseEnhancedConversionData(conversionData) {
+            // Checks if conversion data in custom flags is a stringified object
+            // Conversion data should only be a stringified object or JSON
```

Comment:
It's technically only a JS Object.  There can be a JSON array, and that wouldn't be correct, so I'd be more specific here.
```suggestion
            // Conversion data should only be a stringified object or a JS object 
```